tests/unit/api/test_di.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2021 - 2023 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Testing the di utils from the api subpackage."""
+
+from typing import Annotated
+
+import pytest
+from fastapi import Depends, FastAPI
+from fastapi.testclient import TestClient
+
+from ghga_service_commons.api.di import DependencyDummy
+
+dummy_dependency = DependencyDummy("dummy")
+
+
+def get_test_app() -> FastAPI:
+    """Get a FastAPI app that uses a DependencyDummy."""
+    app = FastAPI()
+
+    @app.get("/")
+    def get_dummy(dummy: Annotated[str, Depends(dummy_dependency)]):
+        """Dummy view function that uses a DependencyDummy."""
+        return dummy
+
+    return app
+
+
+def test_dependency_dummy_no_override():
+    """Test that using a DependencyDummy in a FastAPI app raises an error if it is not
+    overridden.
+    """
+    app = get_test_app()
+    client = TestClient(app)
+    with pytest.raises(RuntimeError):
+        client.get("/")
+
+
+def test_dependency_dummy_override():
+    """Test that using a DependencyDummy in a FastAPI app does not raise an error if it
+    is overridden.
+    """
+    app = get_test_app()
+    app.dependency_overrides[dummy_dependency] = lambda: "dummy"
+    client = TestClient(app)
+    response = client.get("/")
+
+    assert response.json() == "dummy"
```

Comment:
You could also test the string representation of the dummy here.

Reply:
This would not be ideal here, since it goes through the serialization at the server and the client which has nothing to do with the actual injection. But if you like to check the type, I have a suggestions in [69fbaf1](https://github.com/ghga-de/ghga-service-commons/pull/23/commits/69fbaf1f2820fa91934773d97ab14f8ad4708450).

Reply:
Yes, that's also good. But what I meant was simply this:

```python
def test_dependency_dummy_repr():
    """Test the string representation of a DependencyDummy."""
    assert repr(dummy_dependency) == "DependencyDummy(dummy)"
```

Reply:
Ah OK, you have really high standards for testing.

Reply:
Done

Reply:
100% coverage :)